doc/vrrp/VRRP_Adaptation_HLD.md

```diff
@@ -0,0 +1,739 @@
+3) "priority"
+4) "100"
+5) "adv_interval"
+6) "1000"
+7) "admin_status"
+8) "up"
+9) "version"
+10)"3"
+11)"preempt"
+12)"enabled"
+13)"track_interface"
+14)"Ethernet5, Ethernet9"
+15)"priority_decrement"
+16)"20"
+```
+
+#### APPL_DB Changes
+
+INTF_TABLE
+
+Producer: vrrpmgrd and vrrpsyncd
+
+Consumer: intforch
+
+Description: This is an existing table entry that VRRP will extend.
+```
+; holds the VRRP state and VMAC information
+
+key           = INTF_TABLE:Vrrp_intf_name
+                Vrrp_intf_name        ; macvlan device name as a string. The IPv4 format is "Vrrp4-"+"Vrid", while the IPv6 format is "Vrrp6-"+"Vrid"
+
+; field = value
+parent_intf   = interface_name       ; Interface name string. Vlan, Ethernet, PortChannel or sub-interface.
+oper_state    = "up"|"down"          ; The oper state of the Macvlan device in kernel, determines the Master/Backup state of the VRRP instance
+```
+```
+; holds the VRRP Vip information
+
+key           = INTF_TABLE:Vrrp_intf_name:vip
+                Vrrp_intf_name        ; macvlan device name as a string. The IPv4 format is "Vrrp4-"+"Vrid", while the IPv6 format is "Vrrp6-"+"Vrid"
+                vip                   ; Virtual IP address. IPv4 format is x.x.x.x/32, IPv6 format is xxx::xx/128
+
+; field = value
+```
+Example:-
+```
+admin@sonic:~$ redis-cli -n 0 HGETALL "INTF_TABLE:Vrrp4-5"
+
+1) "parent_intf"
+2) "Ethernet51"
+3) "oper_state"
+4) "up"
+
+admin@sonic:~$ redis-cli -n 0 HGETALL "INTF_TABLE:Vrrp4-5:10.10.10.1/32"
+
+1) "NULL"
+2) "NULL"
+```
+
+#### ASIC_DB Changes
+
+Example:-
+```
+admin@sonic:~$ redis-cli -n 1 hgetall "ASIC_STATE:SAI_OBJECT_TYPE_ROUTER_INTERFACE:oid:0x6000000000a80"
+ 1) "SAI_ROUTER_INTERFACE_ATTR_VIRTUAL_ROUTER_ID"
+ 2) "oid:0x300000000003a"
+ 3) "SAI_ROUTER_INTERFACE_ATTR_SRC_MAC_ADDRESS"
+ 4) "00:00:5E:00:01:06"
+ 5) "SAI_ROUTER_INTERFACE_ATTR_TYPE"
+ 6) "SAI_ROUTER_INTERFACE_TYPE_PORT"
+ 7) "SAI_ROUTER_INTERFACE_ATTR_PORT_ID"
+ 8) "oid:0x1000000000025"
+ 9) "SAI_ROUTER_INTERFACE_ATTR_IS_VIRTUAL"
+10) "true"
+```
+
+#### Modules Design and Flows
+
+##### ADD/DEL VRRP instance
+
+![VRRP ADD/DEL VRRP instance Flows](images/VRRP_Config_Instance_Flow.png)
+
+vrrpmgrd:
+  - Listens to VRRP create, delete and parameter change in CONFIG DB
+  - Upon change
+    - Add/del VRRP instance corresponding Macvlan device to kernel with IPs and state.
+    - Updates VRRP instance configuration to the APPL DB, such as Macvlan device name and Vip.
+    - Update changes to vrrpd by using vtysh commands
+
+vrrpd:
+  - Listens to VRRP create, delete and parameter change in vrrpcfgd
+  - Run VRRP state machine
+    - For VRRP instances state transition to Master, set the Macvlan device into protodown off
+    - For VRRP instances state transition to non-Master, set the Macvlan interface into protodown on
+
+vrrpsyncd:
+  - Listens to Macvlan interface programming in kernel. 
+  - Update the kernel Macvlan device's state to the INTF_table entry of APPL DB.
+
+intforch:
```

Comment:
Use vrrpsyncd and vrrporchd sections of VRRP_HLD

Reply:
@philo-micas BRCM has met with Micas and agreed on a converged design on VRRP, please reference the slide deck and update the HLD based our last meeting. I am hoping to get another round of review from the community after your HLD update

Reply:
OK,thanks